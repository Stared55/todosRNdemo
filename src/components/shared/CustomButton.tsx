import * as React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import { Colors } from 'src/utils/styles/colors';
import { yScale } from 'src/utils/styles/scale';
import { ViewStyles, TextStyles } from 'src/utils/types/styleTypes';

export interface CustomButtonProps {
  handler: () => void;
  label: string;
  containerStyles?: ViewStyles;
  labelStyles?: TextStyles;
}

const CustomButton: React.SFC<CustomButtonProps> = ({
  handler,
  label,
  labelStyles,
  containerStyles,
}) => {
  return (
    <TouchableOpacity
      style={{ ...styles.buttonWrapper, ...containerStyles }}
      onPress={handler}>
      <Text style={{ ...styles.text, ...labelStyles }}>{label}</Text>
    </TouchableOpacity>
  );
};

export default CustomButton;

const styles = StyleSheet.create({
  buttonWrapper: {
    alignItems: 'center',
    paddingVertical: yScale(12),
    justifyContent: 'center',
    backgroundColor: Colors.blue,
    borderWidth: 1,
    borderColor: Colors.dark,
    width: '100%',
    marginVertical: yScale(3),
  },
  text: {
    color: Colors.white,
  },
});
