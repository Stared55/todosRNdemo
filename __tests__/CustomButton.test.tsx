import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import CustomButton from 'src/components/shared/CustomButton';

describe('CustomButton', () => {
  const label = 'Label';
  const containerStyles = { backgroundColor: 'red' };
  const labelStyles = { color: 'red' };
  const handler = jest.fn();
  let component: ShallowWrapper;
  let props: any;

  beforeEach(() => {
    component = shallow(
      <CustomButton
        labelStyles={labelStyles}
        containerStyles={containerStyles}
        handler={handler}
        label={label}
      />,
    );
    props = component.props();
    jest.clearAllMocks();
  });

  it('should match to snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should display correct label', () => {
    const labelValue = component.find({ children: label });
    expect(labelValue).toHaveLength(1);
  });

  it('should run callback', () => {
    component.simulate('press');
    expect(handler).toHaveBeenCalledTimes(1);
  });

  it('should render correct container styles', () => {
    expect(props.style.backgroundColor).toEqual('red');
  });

  it('should render correct label styles', () => {
    const children = component.children();
    const childrenProps = children.props();
    expect(childrenProps.style.color).toEqual('red');
  });
});
