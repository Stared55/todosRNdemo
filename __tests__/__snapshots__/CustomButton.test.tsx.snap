// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton should match to snapshot 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderColor": "#2d3436",
      "borderWidth": 1,
      "justifyContent": "center",
      "marginVertical": 5.885294117647058,
      "paddingVertical": 23.541176470588233,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "red",
      }
    }
  >
    Label
  </Text>
</ForwardRef>
`;
